lib/inspec/secrets/yaml.rb

```diff
@@ -9,15 +9,19 @@ class YAML < Inspec.secrets(1)
     attr_reader :attributes
 
     def self.resolve(target)
-      unless target.is_a?(String) && File.file?(target) && target.end_with?('.yml')
```

Comment:
This class is part of our plugin system. Since there can be multiple plugins for a particular area of the InSpec codebase (in this case, the "secrets" area, or rather, the "attributes" area), we need a way to determine whether a plugin is appropriate for the target/file in question.  Therefore, for each plugin, there needs to be a `resolve` method that can be called to determine if this plugin is appropriate for the file.

If we remove the file extension check, this YAML plugin will get called for every file, even if it's not a YAML file. This doesn't matter right now since we only have one "secrets" plugin, but we need to fix it regardless.

My recommendation would be to change this line to something like this:

```
unless target.is_a?(String) && File.file?(target) && [ '.yml', '.yaml' ].include?(File.extname(target).downcase)
```